apps/cookbook/src/app/examples/modal-example/embedded-modal-example/embedded-modal-example.component.ts

```diff
@@ -77,6 +78,34 @@ export class EmbeddedModalExampleComponent implements OnInit {
         setTimeout(() => (this.isLoadingAdditionalContent = false), 2000);
       }
     }
+
+    if (this.alertBeforeClose) {
+      this.modal.canDismiss = () => this.canDismiss();
+    }
+  }
+
+  private canDismiss() {
+    if (!this.alertBeforeClose) return true;
+
+    const config: AlertConfig = {
+      title: 'Are you sure you want to close?',
+      okBtn: 'Yes',
+      cancelBtn: 'Take me back',
+      icon: {
+        name: 'warning',
+        themeColor: 'warning',
+      },
+    };
+
+    return config;
+  }
+
+  _toggleAlertBeforeClose(checked) {
+    this.alertBeforeClose = checked;
+
+    if (checked && !this.modal.canDismiss) {
+      this.modal.canDismiss = () => this.canDismiss();
+    }
```

Comment:
Consider simplifying - we already have a guard in the `canDismiss` method, so we don't need to apply that check 2 times more:
```suggestion
    this.modal.canDismiss = () => this.canDismiss();
  }

  private canDismiss() {
    if (!this.alertBeforeClose) return true;

    const config: AlertConfig = {
      title: 'Are you sure you want to close?',
      okBtn: 'Yes',
      cancelBtn: 'Take me back',
      icon: {
        name: 'warning',
        themeColor: 'warning',
      },
    };

    return config;
  }

  _toggleAlertBeforeClose(checked) {
    this.alertBeforeClose = checked;
```